src/wasm-linker.h

```diff
@@ -47,12 +47,20 @@ class LinkerObject {
     Relocation(Kind kind, uint32_t* data, Name symbol, int addend) :
         kind(kind), data(data), symbol(symbol), addend(addend) {}
   };
+  struct SymbolAlias {
+    Name symbol;
+    Relocation::Kind kind;
+    off_t offset;
+    size_t size;
```

Comment:
`off_t` and `size_t` are types that are meaningful for the host, whereas we want these types to be correct for the target (for example if compiling for wasm32 on a 64-bit machine we could use 32-bit quantities but these would be 64). Also `off_t` is for file offsets, not memory offsets.
For the size I guess we should use `wasm::Address` which is an unsigned of the same size as the address size. For `offset` I guess we want something akin to `ptrdiff_t` but we don't have that. For now, since we actually only support wasm32, maybe we should just  put a `typedef Offset int32_t` below the definition of `Address` in wasm.h. There are definitely places where I used `int` before that should be this offset type instead, but you don't have to fix those in this PR. Let's just avoid making things any worse here :)